langworld_db_data/adders/listed_value_adder.py

```diff
@@ -59,11 +84,47 @@ def _add_to_inventory_of_listed_values(
                     f"Row {row} already contains value you are trying to add"
                 )
 
-            # Keep updating those with each row. This means that at the last row of the
-            # feature they will reach the required values.
-            index_of_last_row_for_given_feature = i
             last_digit_of_value_id = int(row["id"].split(ID_SEPARATOR)[-1])
-            id_of_new_value = feature_id + ID_SEPARATOR + str(last_digit_of_value_id + 1)
+            values_diapason.append(
+                {
+                    "index": last_digit_of_value_id,
+                    "row": i,
+                }
+            )
+
+        last_index_in_feature = values_diapason[-1]["index"]
+        if index_to_assign > last_index_in_feature:
+            raise ListedValueAdderError(
+                f"ID {feature_id + str(index_to_assign)} exceeds the maximal ID {last_index_in_feature}"
+            )
+
+        id_of_new_value = f"{feature_id}{ID_SEPARATOR}{values_diapason[-1]['index'] + 1}"
```

Comment:
I think that this code will benefit from brief, concise comments indicating your intent and stages of the algorithm. I find myself thinking too much about what your code is doing in the following lines

Reply:
Added some comments to the code